src/functional

```diff
@@ -0,0 +1,57 @@
+#!/usr/bin/env php
+<?php
+
+declare(strict_types=1);
+
+use Keboola\Temp\Temp;
+use Symfony\Component\Filesystem\Filesystem;
+use Symfony\Component\Finder\Finder;
+use Symfony\Component\Process\Process;
+
+require_once "vendor/autoload.php";
+
+if ($argc !== 2) {
+    print 'Need tests directory as first param' . PHP_EOL;
+    exit(2);
+}
+
+
+$testFolder = array_pop($argv);
+
+$finder = new Finder();
+$finder->directories()->sortByName()->in($testFolder)->depth(0);
+if (count($finder) === 0) {
+    print 'No tests to run' . PHP_EOL;
+    exit(2);
+}
+
+foreach ($finder as $testSuite) {
+    print "Test " . $testSuite->getPathname() . "\n";
+    $temp = new Temp("my-component");
+    $temp->initRunFolder();
+
+    $fs = new Filesystem();
+    $fs->mirror($testSuite->getPathname() . '/source/data', $temp->getTmpFolder());
+
+    if (!file_exists($temp->getTmpFolder() . "/in/tables")) {
+        $fs->mkdir($temp->getTmpFolder() . "/in/tables", 0777);
+    }
+    if (!file_exists($temp->getTmpFolder() . "/in/files")) {
+        $fs->mkdir($temp->getTmpFolder() . "/in/files", 0777);
+    }
+
+    $fs->mkdir($temp->getTmpFolder() . "/out/tables", 0777);
+    $fs->mkdir($temp->getTmpFolder() . "/out/files", 0777);
+
+    $runCommand = "KBC_DATADIR={$temp->getTmpFolder()} php /code/src/run.php";
+    $runProcess = new Process($runCommand);
+    $runProcess->mustRun();
+
+    $diffCommand = "diff --exclude=.gitkeep --ignore-all-space --recursive " . $testSuite->getPathname() . "/expected/data/out " . $temp->getTmpFolder() . "/out";
+    $diffProcess = new Process($diffCommand);
+    $diffProcess->run();
+    if ($diffProcess->getExitCode() > 0) {
```

Comment:
dost intenzivne je tu potreba i pro failujici testy https://github.com/keboola/processor-iconv/blob/master/tests/functional/run.php#L36 (ten iconv je odnekud opsany a nevim ted odkud)

Reply:
jj, souhlas, debilní je, že to už nekontroluje kvůli čemu to spadlo